pkg/virt-handler/vm_test.go

```diff
@@ -1465,66 +1471,200 @@ var _ = Describe("VirtualMachineInstance", func() {
 			}).Return(vmi, nil)
+
 			controller.Execute()
 		})
 
-		It("should only report the pod network in status", func() {
-			vmi := NewPendingVirtualMachine("testvmi")
-			vmi.Status.Phase = v1.Scheduling
-			pod := NewPodForVirtualMachine(vmi, k8sv1.PodRunning)
-
-			addVirtualMachine(vmi)
-			podFeeder.Add(pod)
-
-			networkName := "test net"
+		It("Should update existing interface with latest IP", func() {
+			vmi := v1.NewMinimalVMI("testvmi")
+			vmi.UID = vmiTestUUID
+			vmi.ObjectMeta.ResourceVersion = "1"
+			interface_name := "interface_name"
 			vmi.Spec.Networks = []v1.Network{
-				v1.Network{
-					Name: networkName,
-					NetworkSource: v1.NetworkSource{
-						Pod: &v1.PodNetwork{
-							VMNetworkCIDR: "1.1.1.1",
-						},
+				{
+					Name:          interface_name,
+					NetworkSource: v1.NetworkSource{Pod: &v1.PodNetwork{}},
+				},
+			}
+			vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{
+				{
+					Name: interface_name,
+					InterfaceBindingMethod: v1.InterfaceBindingMethod{
+						Masquerade: &v1.InterfaceMasquerade{},
 					},
 				},
-				v1.Network{
-					Name: networkName,
-					NetworkSource: v1.NetworkSource{
-						Multus: &v1.MultusNetwork{
-							NetworkName: "multus",
+			}
+			vmi.Status.Phase = v1.Running
+			vmi.Status.Interfaces = []v1.VirtualMachineInstanceNetworkInterface{
+				{
+					Name: interface_name,
+					IP:   "1.1.1.1",
+				},
+			}
+			vmi = addActivePods(vmi, podTestUUID, host)
+
+			podCacheInterface := network.PodCacheInterface{
+				Iface: &v1.Interface{
+					Name: interface_name,
+				},
+				PodIP: "2.2.2.2",
+			}
+			podJson, err := json.Marshal(podCacheInterface)
+			Expect(err).ToNot(HaveOccurred())
+			err = os.MkdirAll(fmt.Sprintf(util.VMIInterfaceDir, vmi.UID), 0755)
+			Expect(err).ToNot(HaveOccurred())
+			vmiInterfacepath := fmt.Sprintf(util.VMIInterfacepath, vmi.UID, interface_name)
+			f, err := os.Create(vmiInterfacepath)
+			Expect(err).ToNot(HaveOccurred())
+			_, err = f.WriteString(string(podJson))
+			Expect(err).ToNot(HaveOccurred())
+			f.Close()
+
+			mockWatchdog.CreateFile(vmi)
+			domain := api.NewMinimalDomainWithUUID("testvmi", vmiTestUUID)
+			domain.Status.Status = api.Running
+			MAC := "1C:CE:C0:01:BE:E7"
+			domain.Spec.Devices.Interfaces = []api.Interface{
+				{
+					MAC:   &api.MAC{MAC: MAC},
+					Alias: &api.Alias{Name: interface_name},
+				},
+			}
+			domain.Status.Interfaces = []api.InterfaceStatus{
+				{
+					Name: interface_name,
+					Mac:  MAC,
+					Ip:   "1.1.1.1",
+					IPs:  []string{"1.1.1.1"},
+				},
+			}
+			vmiFeeder.Add(vmi)
+			domainFeeder.Add(domain)
+
+			updatedVMI := vmi.DeepCopy()
+			updatedVMI.Status.Phase = v1.Running
+			updatedVMI.Status.Conditions = []v1.VirtualMachineInstanceCondition{
+				{
+					Type:   v1.VirtualMachineInstanceIsMigratable,
+					Status: k8sv1.ConditionTrue,
+				},
+			}
+			updatedVMI.Status.MigrationMethod = v1.LiveMigration
```

Comment:
Please explain why is this needed for the test to pass?
This test should be similar to the [non-maquerade ip test](https://github.com/kubevirt/kubevirt/pull/3488/commits/32e0cabfe7c7048a6e2147cbadfc8cb296647312#diff-14837489e428608f8ed1f61f9d7242bcR1713).
Please notice the non-masquerade test is very specific to network and to the purpose of the test which is verifying the ip on the status. 

You should change `vmiInterface.EXPECT().Update(updatedVMI)` to `vmiInterface.EXPECT().Update(gomock.Any())` and check the arguments related to the `status.interfaces`(which you already do).
This way, you won't need to care about the other values on the updatedVmi (you won't even need the `updatedVmi`).

Reply:
Done